### DISCUSSION TITLE
Cropping the y-vector

### DISCUSSION THREAD

**Discussion by AdarshMJ** [pyg-team/pytorch_geometric#4784] [Q&A]
Hi @rusty1s 

My graph dataset looks like this. - `DataBatch(x=[1569, 79], edge_index=[2, 3406], edge_attr=[3406, 10], y=[64, 3393], batch=[1569], ptr=[65])`

I want to clip my y vector, that is instead of considering 3393 values, I want to consider only 1000 of its values based on some criteria, like for example -

```
for graph in dataset:
   if torch.sum(graph.y[1000:-1]) > 0.5: 
     graph.y.remove
   else: 
     graph.y=graph.y[0:1000]
```
How do I remove these samples? Any suggestions would be helpful. Thank you!

**Comment by rusty1s** [comment]
I might be misunderstanding this, but can't you simply do
```python
data_list = []
for graph in dataset:
   if not torch.sum(graph.y[1000:-1]) > 0.5: 
     graph.y=graph.y[0:1000]
     data_list.append(graph)
```

**Reply by AdarshMJ** [reply]
I did something like this. -
```Python
for graph in dataset :
  if(torch.sum(graph.y[1000:-1]))>0.5:
        graph.y = graph.y[ :, 1000:-1]
  else:
    graph.y = graph.y[:,0:1000]
```

Thank you for the answer it works :)

**Reply by AdarshMJ** [reply]
Additionally I am trying to predict an output which looks something like this -
![GT](https://user-images.githubusercontent.com/24205200/172814114-9e814336-2462-444a-a581-391e6d99d710.png) , 

so far I have tried using PNA + MSE loss and PNA+Cosine similarity the results are not that great. Do you have any suggestions for this? Perhaps should I try with some kind of graph edit distance kind of loss or something?

**Reply by rusty1s** [reply]
This looks like a pretty imbalanced label distribution. Perhaps re-weighting positive labels help?

**Reply by AdarshMJ** [reply]
Thanks for the quick reply. Could you please elaborate on that? Some kind of reference would be great. This is the prediction I am getting, but as you can see its not that great.

![example_op_Exp1Cosine_100045_100002](https://user-images.githubusercontent.com/24205200/172855388-994fe042-ca1d-4e21-b0c7-3068c22da811.png)
.

**Reply by rusty1s** [reply]
You can look at the `pos_weight` argument of the [`BCEWithLogitsLoss`](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html#torch.nn.BCEWithLogitsLoss). This lets you put more emphasis on the positive/underrepresented labels. I also suggest to clamp the output into range 0 and 1, e.g., via `sigmoid`.
